tests/unit/scheduler.c

```diff
@@ -72,331 +56,221 @@ int verify_schedule(nccl_net_ofi_schedule_t *schedule, nccl_net_ofi_schedule_t *
-		free(ref_schedule);
-		return ret;
-	}
-	ref_schedule->num_xfer_infos = 0;
-	ret = verify_schedule(schedule, ref_schedule);
-	if (ret) {
-		NCCL_OFI_WARN("Verification failed");
-		free(ref_schedule);
-		return ret;
-	}
-	free(schedule);
-
-	/************************/
-	/* Test one rail        */
-	/************************/
-
-	/* No data */
-	size = 0;
-	num_rails = 1;
-	align = 1;
-	ret = create_multiplexed(size, num_rails, align, &schedule);
-	if (ret) {
-		NCCL_OFI_WARN("Failed to create multiplexed schedule");
-		free(ref_schedule);
-		return ret;
-	}
-	ref_schedule->num_xfer_infos = 0;
-	ret = verify_schedule(schedule, ref_schedule);
-	if (ret) {
-		NCCL_OFI_WARN("Verification failed");
-		free(ref_schedule);
-		return ret;
-	}
-	free(schedule);
-
-	/* Data size = align - 1 */
-	size = 1;
-	num_rails = 1;
-	align = 2;
-	ret = create_multiplexed(size, num_rails, align, &schedule);
-	if (ret) {
-		NCCL_OFI_WARN("Failed to create multiplexed schedule");
-		free(ref_schedule);
-		return ret;
-	}
-	ref_schedule->num_xfer_infos = 1;
-	ref_schedule->rail_xfer_infos[0].rail_id = 0;
-	ref_schedule->rail_xfer_infos[0].offset = 0;
-	ref_schedule->rail_xfer_infos[0].msg_size = size;
-	ret = verify_schedule(schedule, ref_schedule);
-	if (ret) {
-		NCCL_OFI_WARN("Verification failed");
-		free(ref_schedule);
-		return ret;
+	if (!(*schedule)) {
+		NCCL_OFI_WARN("Could not allocate schedule");
+		return -ENOMEM;
 	}
-	free(schedule);
+	(*schedule)->num_xfer_infos = num_xfer_infos;
 
-	/* Data size = align */
-	size = 2;
-	num_rails = 1;
-	align = 2;
-	ret = create_multiplexed(size, num_rails, align, &schedule);
-	if (ret) {
-		NCCL_OFI_WARN("Failed to create multiplexed schedule");
-		free(ref_schedule);
-		return ret;
-	}
-	ref_schedule->num_xfer_infos = 1;
-	ref_schedule->rail_xfer_infos[0].rail_id = 0;
-	ref_schedule->rail_xfer_infos[0].offset = 0;
-	ref_schedule->rail_xfer_infos[0].msg_size = size;
-	ret = verify_schedule(schedule, ref_schedule);
-	if (ret) {
-		NCCL_OFI_WARN("Verification failed");
-		free(ref_schedule);
-		return ret;
-	}
-	free(schedule);
+	return ret;
+}
 
-	/* Data size = align + 1 */
-	size = 3;
-	num_rails = 1;
-	align = 2;
-	ret = create_multiplexed(size, num_rails, align, &schedule);
-	if (ret) {
-		NCCL_OFI_WARN("Failed to create multiplexed schedule");
-		free(ref_schedule);
-		return ret;
-	}
-	ref_schedule->num_xfer_infos = 1;
-	ref_schedule->rail_xfer_infos[0].rail_id = 0;
-	ref_schedule->rail_xfer_infos[0].offset = 0;
-	ref_schedule->rail_xfer_infos[0].msg_size = size;
-	ret = verify_schedule(schedule, ref_schedule);
-	if (ret) {
-		NCCL_OFI_WARN("Verification failed");
-		free(ref_schedule);
-		return ret;
-	}
-	free(schedule);
+int set_ref_schedule(nccl_net_ofi_schedule_t *schedule, int index, int rail_id, int offset, int msg_size) {
+	int ret = 0;
+    if (index >= schedule->num_xfer_infos) {
+		NCCL_OFI_WARN("Index out of bounds");
+		return -EINVAL;
+    }
```

Comment:
In case you push to this branch again... indenting is still off here.

Reply:
Done

Reply:
arun, for future reference, take a look at this clang-format file:

https://gist.github.com/aws-nslick/595ce0a9a66452a8323f84ff8337c359

You can set it up for your source tree on the headnode for the samwise cluster with the following:

```
mkdir -p ~/.config/git/ && echo ".clang-format" >> ~/.config/git/global_ignores
git config --global core.excludesFile ~/.config/git/global_ignores
cd /path/to/aws-ofi-nccl
wget "https://gist.githubusercontent.com/aws-nslick/595ce0a9a66452a8323f84ff8337c359/raw/ad3fc5433d0974dc335dcda677493fe0628d0cf6/gistfile1.txt" -O .clang-format
echo -e '#!/bin/sh\ngit clang-format-19' > .git/hooks/pre-commit
chmod +x .git/hooks/pre-commit
```

whenever you have staged changes and you go to `git commit`, this hook will run over the files you've changed in your diff, only over the lines you've changed, and it will create unstaged changes that applies the styling for anywhere that you've deviated. If you haven't deviated at all, it will pop you into the commit message window. If you deviated, it will tell you that there are new unstaged changes.

If you like the "fixes", you can just stage them and rerun `git commit`. 

If you don't like them or it gets something slightly wrong, just discard the hunks or check the files out. Then run `git commit --no-verify` to skip running the formatting and to just commit as-is.

You can also use my .editorconfig file: 

```
echo ".editorconfig" >> ~/.config/git/global_ignores
cd /path/to/aws-ofi-nccl
wget "https://gist.githubusercontent.com/aws-nslick/c52b2a54e30a121e7146a2089b2be0cc/raw/2527bdbc3107ee32a10a89978ca73ac64729f116/gistfile1.txt" -O .editorconfig
```

If you reload vscode/sublime/emacs/neovim/whatever, it will pick up the settings defined there and make sure that tab inserts tabs and not spaces, and that the width is treated as 8. 

There's a few other goodies on https://github.com/aws-nslick/nccl-net-ofi/tree/qol if you're interested.

Reply:
Thanks a ton for the details instructions. I will setup clang-format for my code-repo with the .editorconfig you have shared. Will format my code and post another revision. Thanks